stdlib/public/core/Random.swift

```diff
@@ -94,6 +94,7 @@ extension RandomNumberGenerator {
     upperBound: T
   ) -> T {
     _precondition(upperBound != 0, "upperBound cannot be zero.")
+#if arch(i386) || arch(arm)
```

Comment:
Please add a TODO here and a swift bug to remove this when no longer needed.

Reply:
[SR-10912](https://bugs.swift.org/browse/SR-10912) Use Lemire's Nearly Divisionless Random Integer Generation on 32-bit Systems